src/org/usfirst/frc/team467/robot/OpticalSensor.java

```diff
@@ -0,0 +1,26 @@
+package org.usfirst.frc.team467.robot;
+import edu.wpi.first.wpilibj.DigitalInput;
+
+public class OpticalSensor {
+private final static int OPTICAL_CHANNEL = 5;
+public DigitalInput di;
+private static OpticalSensor instance;
+public boolean gotCube = false;
+	
+	private OpticalSensor() {
+		di = new DigitalInput(OPTICAL_CHANNEL);
+	}
+	
+	public static OpticalSensor getInstance() {
+		if (instance == null) {
+			instance = new OpticalSensor();
+		}
+		
+		return instance;
+	}
+	
+	public boolean hasCube() {
```

Comment:
This should be renamed to "canSeeTarget" or similar. It shouldn't know that you're looking at a cube, that is a detail known by the Grabber class, but not here.